soh/src/overlays/actors/ovl_En_Kakasi2/z_en_kakasi2.c

```diff
@@ -118,10 +118,12 @@ void func_80A90264(EnKakasi2* this, GlobalContext* globalCtx) {
 
     this->unk_194++;
 
-    if (((BREG(1) != 0) || ((CVar_GetS32("gSkipScarecrow", 0) != 0) && 
-            (globalCtx->msgCtx.ocarinaAction == OCARINA_ACTION_FREE_PLAY) && gSaveContext.scarecrowSpawnSongSet)) &&
-            (this->actor.xzDistToPlayer < this->maxSpawnDistance.x) &&
-            (fabsf(player->actor.world.pos.y - this->actor.world.pos.y) < this->maxSpawnDistance.y)){
+    bool skipScarecrow =
+        (CVar_GetS32("gSkipScarecrow", 0) && gSaveContext.scarecrowSpawnSongSet && globalCtx->msgCtx.ocarinaAction == OCARINA_ACTION_FREE_PLAY) ||
+        (gSaveContext.n64ddFlag && Randomizer_GetSettingValue(RSK_SKIP_SCARECROWS_SONG) && globalCtx->msgCtx.ocarinaAction == OCARINA_ACTION_FREE_PLAY);
```

Comment:
This is the only thing I'm not sure about how to do better.

I had this before:
```C
bool skipScarecrow =
        globalCtx->msgCtx.ocarinaAction == OCARINA_ACTION_FREE_PLAY &&
        (CVar_GetS32("gSkipScarecrow", 0) && gSaveContext.scarecrowSpawnSongSet) ||
        (gSaveContext.n64ddFlag && Randomizer_GetSettingValue(RSK_SKIP_SCARECROWS_SONG));
```
But that still resolved to true even without `globalCtx->msgCtx.ocarinaAction == OCARINA_ACTION_FREE_PLAY` being true when one of the other if statements was true. Is this something about declaring something as a bool that works differently than having them in an if statement?

Reply:
just looking at it i think the issue is it was reading it as

```cpp
(freePlay && skipAndSpawn) || (theRest)
```

instead of

```cpp
freePlay && (skipAndSpawn || theRest)
```

Reply:
Is it fine to leave it as it is then? I tried to make it work with only declaring it once, but whatever I tried didn't work.

Reply:
did you try
```cpp
bool skipScarecrow =
        globalCtx->msgCtx.ocarinaAction == OCARINA_ACTION_FREE_PLAY &&
        ((CVar_GetS32("gSkipScarecrow", 0) && gSaveContext.scarecrowSpawnSongSet) ||
        (gSaveContext.n64ddFlag && Randomizer_GetSettingValue(RSK_SKIP_SCARECROWS_SONG)));
```

Reply:
Nope, but in hindsight I really should've. I'll try that in a bit and test it, but I'm actually pretty sure that'll actually work.

Reply:
That totally worked, woops.. Thanks for the help.